src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/Logging/RoslynLogger.cs

```diff
@@ -101,8 +102,11 @@ public void Log(FunctionId functionId, LogMessage logMessage)
                 return;
             }
 
+            using var pooledObject = s_propertyPool.GetPooledObject();
```

Comment:
💡 If this array frequently has 0-4 members at the end, consider using `TemporaryArray<Property>` instead of `List<Property>`. This will avoid heap allocations _and_ churn in the array pool when fewer than 5 properties are added.

Reply:
> 💡 If this array frequently has 0-4 members at the end, consider using `TemporaryArray<Property>` instead of `List<Property>`. This will avoid heap allocations _and_ churn in the array pool when fewer than 5 properties are added.

I was under the impression that the assembly this was being defined had extremely minimal dependencies and thus I couldn't use that type, but I talked with David and that's evidently not the case, so I'll give that a go!

Reply:
Turns out it's the devkit assembly which has the limited dependencies, and that assembly holds an implementation of this interface. Even though that project manually includes the TemporaryArray.cs file, the implementation can't reference the one from MS.CA.dll, so it appears for now, TemporaryArray isn't an option.

Reply:
Interested in this group's thoughts on whether the ITelemetryReporter interface should just take in an interface (IEnumerable/IReadOnlyList) instead of a concrete type (List). My primary intent with this PR is to get rid of ImmutableDictionary in this codepath and reduce allocations, both of which are satisfied using either the concrete type or one of these interfaces. ITelemetryReporter isn't public, so leaking out the concrete type isn't horrible, but it still feels a little wrong to me.

Reply:
List is better, since it can be enumerated cheaply.  Given how much these telemetry functions are called, we should make them as lean and mean as possible.